helios-core/src/main/kotlin/helios/core/errors.kt

```diff
@@ -1,10 +1,19 @@
 package helios.core
 
-sealed class DecodingError
-data class StringDecodingError(val value: Json) : DecodingError()
-data class BooleanDecodingError(val value: Json) : DecodingError()
-data class NumberDecodingError(val value: Json) : DecodingError()
-data class ArrayDecodingError(val value: Json) : DecodingError()
-data class ObjectDecodingError(val value: Json) : DecodingError()
-data class KeyNotFound(val name: String) : DecodingError()
-data class EnumValueNotFound(val value: Json) : DecodingError()
\ No newline at end of file
+import arrow.core.Option
+import java.time.format.DateTimeFormatter
+
+interface DecodingError
```

Comment:
This shouldn't be necessary with a generic ADT case `ExceptionOnDecoding`. We should keep this as an ADT instead of an open hierarchy.

Reply:
I thought that people creating their custom codecs could want to have their own errors but if we are sure we are covering enough possibilities I agree with that :+1:

Reply:
With `ExceptionOnDecoding` users can match on that for custom errors.
With an interface they can extend but they cannot match due to the open inheritance, so the `sealed` hierarchy is  preferred IMO.